kimchi/src/circuits/expr.rs

```diff
@@ -540,8 +551,148 @@ pub enum Expr<C> {
     UnnormalizedLagrangeBasis(i32),
     Pow(Box<Expr<C>>, u64),
     Cache(CacheId, Box<Expr<C>>),
-    /// Expression is conditional on the given feature flag, returns 0 if disabled.
-    EnabledIf(FeatureFlag, Box<Expr<C>>),
+    /// If the feature flag is enabled, return the first expression; otherwise, return the second.
+    IfFeature(FeatureFlag, Box<Expr<C>>, Box<Expr<C>>),
+}
+
+impl<C: Zero + One + Neg<Output = C> + PartialEq + Clone> Expr<C> {
+    fn apply_feature_flags_inner(&self, features: &FeatureFlags) -> (Expr<C>, bool) {
```

Comment:
What is the boolean trying to represent? Whether it is a fully reduced expression?